### PR TITLE
Polish Secrets Manager Integration

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Correct handling `optional` config data and `failFast` for old type of integration.

Fixes logging in `AwsSecretsManagerConfigDataLocationResolver`.

Handles the case when `AwsSecretsManagerPropertySources` returns`null`.

## :bulb: Motivation and Context

Follow up to #721.

## :green_heart: How did you test it?

Unit & integration tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated reference documentation to reflect the change
- [x] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: docs/src/main/asciidoc/_configprops.adoc [hunk 1]
```diff
@@ -1,67 +1,68 @@
 |===
 |Name | Default | Description
 
-|aws.paramstore.default-context | application | 
-|aws.paramstore.enabled | true | Is AWS Parameter Store support enabled.
+|aws.paramstore.default-context | `application` |
```
Comment (eddumelendez) [inline docs/src/main/asciidoc/_configprops.adoc:4]: changes in this file are not related

File: spring-cloud-aws-secrets-manager-config/src/test/java/org/springframework/cloud/aws/secretsmanager/AwsSecretsManagerPropertySourceLocatorTest.java [hunk 2]
```diff
@@ -127,7 +132,34 @@ public void contextSpecificOrderExpected() {
 		assertThat(contextToBeTested.get(1)).isEqualTo("/secret/messaging-service");
 		assertThat(contextToBeTested.get(2)).isEqualTo("/secret/application_test");
 		assertThat(contextToBeTested.get(3)).isEqualTo("/secret/application");
+	}
+
+	@Test
+	public void whenFailFastIsTrueAndSecretDoesNotExistThrowsException() {
```
Comment (eddumelendez) [inline spring-cloud-aws-secrets-manager-config/src/test/java/org/springframework/cloud/aws/secretsmanager/AwsSecretsManagerPropertySourceLocatorTest.java:138]:
```suggestion
	void whenFailFastIsTrueAndSecretDoesNotExistThrowsException() {
```

File: spring-cloud-aws-secrets-manager-config/src/test/java/org/springframework/cloud/aws/secretsmanager/AwsSecretsManagerPropertySourceLocatorTest.java [hunk 3]
```diff
@@ -127,7 +132,34 @@ public void contextSpecificOrderExpected() {
...
+		when(smClient.getSecretValue(any(GetSecretValueRequest.class))).thenThrow(ResourceNotFoundException.class);
+
+		AwsSecretsManagerPropertySourceLocator locator = new AwsSecretsManagerPropertySourceLocator(smClient,
+				properties);
+		assertThatThrownBy(() -> locator.locate(env))
+				.isInstanceOf(AwsSecretsManagerPropertySourceNotFoundException.class);
+	}
+
+	@Test
+	public void whenFailFastIsFalseAndSecretDoesNotExistReturnsEmptyPropertySource() {
```
Comment (eddumelendez) [inline spring-cloud-aws-secrets-manager-config/src/test/java/org/springframework/cloud/aws/secretsmanager/AwsSecretsManagerPropertySourceLocatorTest.java:151]:
```suggestion
	void whenFailFastIsFalseAndSecretDoesNotExistReturnsEmptyPropertySource() {
```

**Review by eddumelendez — CHANGES_REQUESTED** [review]

**Review by eddumelendez — APPROVED** [review]
still see changes at `docs/src/main/asciidoc/_configprops.adoc` which I think they are not related to this PR.

**Comment by maciejwalkowiak** [comment]
That's me losing the battle with git and intellij
